### ISSUE TITLE
Allow default background color being off-scale

### ISSUE THREAD

**Issue by MightyPork** [espterm/espterm-firmware#74]
<s>however since colors are 4-bit, we have to add this as an attribute to the attrs bitfield, and take care of it when serializing screen to JS string.

one extra bit will fit into Encode2B safely, or we can finally move to binary messages in socket (but there may be a problem with the init XHR?) </s>

colors are now each sent using Encode2B which can handle almost two bytes (but not quite), or 3B when combined (which can handle over 2 bytes). This means we can encode this as an additional bit in the transfer.

the issue is with storage:  all 8 bits of the attrs word are now used. We may need to make a compromise and not hold all attribs at once. 

- The three bits for underline, overline and strike come to mind.
- It probably doesnt make sense to have italic fraktur?
- is fraktur even good for anything other than being a curiosity?? (i like it, but ...) /
- we can even convert to fraktur unicode server/side and free the bit entirely. however this means SGR change commands (added in a later update - those are the rectangle operations from VT5xx) will not be able to unset fraktur. i guess that's not an huge deal. also this meas fraktur copy/paste will have to grab the unicode, not ascii. again probably not super important

- setting foreground offscale is not a common feature and it would require one more bit

so in conclusion

- [ ] convert fraktur server-side and use the former fraktur bit to mean Default Background.
- [ ] handle this DefaultBG bit client-side
- [ ] when the bit is set, we need not send background change sequences to client as it will ignore those
- [ ] make the default bg hex configurable and add server-side storage (and also have a default value associated with a theme, eg. Ambiance is meant to have deep a purple background by default)
- [ ] add the def bg hex to the update message

**Comment by MightyPork** [comment]
alternatively we could give up full rgb hex and just pick from the 256 colors, that makes it just one byte that's a bit easier to handle

**Comment by c-probably-archived20260718** [comment]
the attributes byte could also be extended like UTF-8 to have a bit that, when set, extends it to two bytes

**Comment by MightyPork** [comment]
the issue is with back-end storage, not transport encoding.

With 80x25 that is 2000 bytes for each byte added to the cell struct

Currently, that is:

- 1 byte - character or utf-8 look-up table reference
- 1 byte - fg 0-255
- 1 byte - bg 0-255
- 1 byte - attribute bits

I'd love to cut it further, but have not devised a way to do that yet. your idea with "attribute ranges" would blow up with something like the rainbow demo...

**Comment by c-probably-archived20260718** [comment]
Yeah that'd be like trying to compress RGB noise losslessly

**Comment by MightyPork** [comment]
This is now implemented, details are in #191.
